server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileWsSupport.java

```diff
@@ -19,26 +19,94 @@
+import static org.sonarqube.ws.client.component.ComponentsWsParameters.PARAM_ORGANIZATION;
 
 @ServerSide
 public class QProfileWsSupport {
 
+  private final DbClient dbClient;
   private final UserSession userSession;
   private final DefaultOrganizationProvider defaultOrganizationProvider;
 
-  public QProfileWsSupport(UserSession userSession, DefaultOrganizationProvider defaultOrganizationProvider) {
+  public QProfileWsSupport(DbClient dbClient, UserSession userSession, DefaultOrganizationProvider defaultOrganizationProvider) {
+    this.dbClient = dbClient;
     this.userSession = userSession;
     this.defaultOrganizationProvider = defaultOrganizationProvider;
   }
 
+  public String getOrganizationKey(QualityProfileDto profile, DbSession dbSession) {
+    String organizationUuid = profile.getOrganizationUuid();
+    return dbClient.organizationDao().selectByUuid(dbSession, organizationUuid)
+      .orElseThrow(() -> new IllegalStateException("Cannot load organization with uuid=" + organizationUuid))
+      .getKey();
+  }
+
+  /**
+   * @deprecated provide session
+   */
+  @Deprecated
+  public OrganizationDto getOrganizationByKey(String organizationKey) {
+    try (DbSession dbSession = dbClient.openSession(false)) {
+      return getOrganizationByKey(dbSession, organizationKey);
+    }
+  }
+
+  public OrganizationDto getOrganizationByKey(DbSession dbSession, @Nullable String organizationKey) {
+    String organizationOrDefaultKey = Optional.ofNullable(organizationKey)
+      .orElseGet(defaultOrganizationProvider.get()::getKey);
+    return WsUtils.checkFoundWithOptional(
+      dbClient.organizationDao().selectByKey(dbSession, organizationOrDefaultKey),
+      "No organizationDto with key '%s'", organizationOrDefaultKey);
+  }
+
+  public String getOrganizationKeyByUuid(String organizationUuid) {
+    return getOrganizationByUuid(organizationUuid)
+      .orElseThrow(() -> new IllegalStateException(String.format("Requested the key for non-existing organization uuid '%s'.", organizationUuid)))
+      .getKey();
+  }
+
+  private Optional<OrganizationDto> getOrganizationByUuid(String organizationUuid) {
+    try (DbSession dbSession = dbClient.openSession(false)) {
+      return dbClient.organizationDao()
+        .selectByUuid(dbSession, organizationUuid);
+    }
+  }
+
   public void checkQProfileAdminPermission() {
     userSession
       .checkLoggedIn()
       .checkPermission(ADMINISTER_QUALITY_PROFILES, defaultOrganizationProvider.get().getUuid());
   }
+
+  public NewParam createOrganizationParam(NewAction create) {
+    return create
+    .createParam(PARAM_ORGANIZATION)
+    .setDescription("Organization key")
+    .setRequired(false)
+    .setInternal(true)
+    .setExampleValue("my-org");
+  }
+
+  /**
+   * @deprecated should not be required anymore, once all quality profile webservices are migrated to use organizations.
+   */
+  @Deprecated
+  public OrganizationDto getDefaultOrganization(DbSession dbSession) {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)